### ISSUE TITLE
Harfbuzz fails to build on Ubuntu 20.04 (WSL 2)

### ISSUE THREAD

**Issue by Tyrubias** [Homebrew/linuxbrew-core#20888]
# Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.

- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

I tried updating my exisiting `harfbuzz` installation.

## What happened (include command output)

After `harfbuzz` was patched, building with `ninja` failed

<details>
  <summary>Command output</summary>
  <pre>

  $ brew upgrade

==> Upgrading 1 outdated package:
harfbuzz 2.6.8 -> 2.7.0
==> Upgrading harfbuzz 2.6.8 -> 2.7.0
==> Downloading https://github.com/harfbuzz/harfbuzz/commit/7c61caa7384e9c3afa0d9237bf6cd303eb5ef3a1.patch?full_index=1
Already downloaded: /home/vsong/.cache/Homebrew/downloads/c06e429357a688c79e263d2094e103a4a9b969196ed5bcb6de9e13b6cc9698fd--7c61caa7384e9c3afa0d9237bf6cd303eb5ef3a1.patch
==> Downloading https://github.com/harfbuzz/harfbuzz/raw/fc0daafab0336b847ac14682e581a8838f36a0bf/test/shaping/fonts/sha1sum/270b89df543a7e48e206a2d830c0e10e5265c630.ttf
Already downloaded: /home/vsong/.cache/Homebrew/downloads/b0f20064588dc70dd5c74e36b842326d9dbbeacab35c63c3d900a8b34aaba98f--270b89df543a7e48e206a2d830c0e10e5265c630.ttf
==> Downloading https://github.com/harfbuzz/harfbuzz/archive/2.7.0.tar.gz
Already downloaded: /home/vsong/.cache/Homebrew/downloads/375e0a0893ea76812873fd49ef3bb2d3f178a39933235ffb8393748fbf388f69--harfbuzz-2.7.0.tar.gz
==> Patching
==> Applying 7c61caa7384e9c3afa0d9237bf6cd303eb5ef3a1.patch
patching file src/meson.build
Hunk #1 succeeded at 397 with fuzz 1 (offset -8 lines).
Hunk #2 succeeded at 406 (offset -8 lines).
Hunk #3 succeeded at 430 (offset -8 lines).
Hunk #4 succeeded at 532 (offset -8 lines).
Hunk #5 succeeded at 618 (offset -8 lines).
==> meson --prefix=/home/linuxbrew/.linuxbrew/Cellar/harfbuzz/2.7.0 --libdir=/home/linuxbrew/.linuxbrew/Cellar/harfbuzz/2.7.0/lib --buildtype=release --default-library=both -Dcairo=enabled -Dcoretext=enabled -Dfreetype=ena
==> ninja
Last 15 lines from /home/vsong/.cache/Homebrew/Logs/harfbuzz/02.ninja:
[115/289] Compiling C++ object test/api/test-cplusplus.p/test-cplusplus.cc.o
[116/289] Compiling C object test/api/test-ot-ligature-carets.p/test-ot-ligature-carets.c.o
[117/289] Compiling C object test/api/test-common.p/test-common.c.o
[118/289] Compiling C object test/api/test-map.p/test-map.c.o
[119/289] Generating hb-gobject-enums-tmp.cc with a meson_exe.py custom command
[120/289] Compiling C object test/api/test-buffer.p/test-buffer.c.o
[121/289] Compiling C++ object util/hb-ot-shape-closure.p/options.cc.o
[122/289] Compiling C object test/api/test-font.p/test-font.c.o
[123/289] Compiling C object test/api/test-object.p/test-object.c.o
[124/289] Compiling C++ object src/test-iter.p/test-iter.cc.o
[125/289] Compiling C++ object src/test-bimap.p/test-bimap.cc.o
[126/289] Compiling C++ object src/libharfbuzz.so.0.20700.0.p/hb-ot-layout.cc.o
[127/289] Compiling C++ object src/libharfbuzz-subset.so.0.20700.0.p/hb-subset-plan.cc.o
[128/289] Compiling C++ object src/libharfbuzz-subset.so.0.20700.0.p/hb-subset.cc.o
ninja: build stopped: subcommand failed.

READ THIS: https://docs.brew.sh/Troubleshooting
  </pre>
</details>
    
<details>
<summary>brew config</summary>

<pre>
HOMEBREW_VERSION: 2.4.9-218-g7075959
ORIGIN: https://github.com/Homebrew/brew
HEAD: 7075959919691b9debac3d1bf00b5b257eb036a9
Last commit: 3 hours ago
Core tap ORIGIN: https://github.com/Homebrew/linuxbrew-core
Core tap HEAD: ff8d5de9caecfa4f75406990343cb222ec8e67a5
Core tap last commit: 9 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 16
CPU: 16-core 64-bit skylake
Homebrew Ruby: 2.6.3 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
Clang: N/A
Git: 2.25.1 => /bin/git
Curl: 7.68.0 => /usr/bin/curl
Kernel: Linux 4.19.104-microsoft-standard x86_64 GNU/Linux
OS: Ubuntu 20.04.1 LTS (focal)
Host glibc: 2.31
/usr/bin/gcc: 9.3.0
glibc: N/A
gcc: 5.5.0_7
xorg: 20170115_1
</pre>
</details>

<details>
<summary>brew doctor</summary>
<pre>
Your system is ready to brew.
</pre>
</details>

<details>
<summary>brew gist-logs harfbuzz</summary>
<pre>
https://gist.github.com/Tyrubias/e1d506f78672f29cfeb6ff6029839d39
</pre>
</details>

## What you expected to happen

Building of `harfbuzz` should have succeeded.

## Step-by-step reproduction instructions (by running `brew install` commands)

`brew upgrade harfbuzz`

**Comment by holgr** [comment]
I see the same on CentOS 7 and CentOS 8.

**Comment by iMichka** [comment]
```
glib-2.0/glib/gatomic.h:128:15: error: variable 'gapg_temp_atomic' set but not used [-Werror=unused-but-set-variable]
     gpointer *gapg_temp_atomic = (gpointer *)(atomic);                       \
               ^
```

This is an upstream bug: https://github.com/harfbuzz/harfbuzz/issues/2555

**Comment by maxim-belkin** [comment]
builds with
```
-Dglib=disabled
-Dgobject=disabled
-Dintrospection=disabled
```

**Comment by sqfreak** [comment]
@maxim-belkin 's suggestion gets it to build, but it fails `brew test harfbuzz` for me:
```brew test harfbuzz
Testing harfbuzz
sh: 1: /home/linuxbrew/.linuxbrew/Cellar/harfbuzz/2.7.0/bin/hb-shape: not found
Error: harfbuzz: failed
An exception occurred within a child process:
  Test::Unit::AssertionFailedError: <"[glyph201=0+1183|U0D4D=0+0]"> expected but was
<"">.
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/gems/2.6.0/gems/test-unit-3.2.9/lib/test/unit/assertions.rb:55:in `block in assert_block'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/gems/2.6.0/gems/test-unit-3.2.9/lib/test/unit/assertions.rb:1636:in `_wrap_assertion'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/gems/2.6.0/gems/test-unit-3.2.9/lib/test/unit/assertions.rb:53:in `assert_block'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/gems/2.6.0/gems/test-unit-3.2.9/lib/test/unit/assertions.rb:240:in `assert_equal'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/harfbuzz.rb:61:in `block (2 levels) in <class:Harfbuzz>'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:111:in `block in unpack'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:197:in `block in mktemp'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:57:in `block in run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:57:in `chdir'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:57:in `run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:196:in `mktemp'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:106:in `unpack'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:81:in `stage'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/harfbuzz.rb:59:in `block in <class:Harfbuzz>'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1818:in `block (3 levels) in run_test'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils.rb:474:in `with_env'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1817:in `block (2 levels) in run_test'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:900:in `with_logging'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1816:in `block in run_test'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:2060:in `block in mktemp'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:57:in `block in run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:57:in `chdir'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:57:in `run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:2059:in `mktemp'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1810:in `run_test'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/test.rb:35:in `block in <main>'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/timeout.rb:93:in `block in timeout'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/timeout.rb:33:in `block in catch'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/timeout.rb:108:in `timeout'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/test.rb:34:in `<main>'```

**Comment by berndverst** [comment]
I can confirm this resolved the issue with LinuxBrew (using Ubuntu 20.04 in WSL2).

```
==> Upgrading 1 outdated package:
harfbuzz 2.6.8 -> 2.7.0
==> Upgrading harfbuzz 2.6.8 -> 2.7.0
==> Downloading https://linuxbrew.bintray.com/bottles/harfbuzz-2.7.0.x86_64_linux.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/0f/0fceff4d6205900e499cde14c21836b0a4cc46e18a306b3f2a114dac2ab0b60d?__gd
######################################################################## 100.0%
==> Pouring harfbuzz-2.7.0.x86_64_linux.bottle.tar.gz
Fetching gem metadata from https://rubygems.org/.........
Fetching concurrent-ruby 1.1.7
Fetching minitest 5.14.1
Fetching thread_safe 0.3.6
Installing minitest 5.14.1
Installing thread_safe 0.3.6
Installing concurrent-ruby 1.1.7
Fetching zeitwerk 2.4.0
Installing zeitwerk 2.4.0
Fetching ast 2.4.1
Fetching bindata 2.4.8
Installing ast 2.4.1
Installing bindata 2.4.8
Using bundler 1.17.2
Fetching byebug 11.1.3
Installing byebug 11.1.3 with native extensions
Fetching colorize 0.8.1
Installing colorize 0.8.1
Fetching json 2.3.1
Fetching docile 1.3.2
Installing json 2.3.1 with native extensions
Installing docile 1.3.2
Fetching simplecov-html 0.12.2
Installing simplecov-html 0.12.2
Fetching coderay 1.1.3
Installing coderay 1.1.3
Fetching highline 2.0.3
Installing highline 2.0.3
Fetching connection_pool 2.2.3
Installing connection_pool 2.2.3
Fetching diff-lcs 1.4.4
Installing diff-lcs 1.4.4
Fetching unf_ext 0.0.7.7
Installing unf_ext 0.0.7.7 with native extensions
Fetching hpricot 0.8.6
Installing hpricot 0.8.6 with native extensions
Fetching mime-types-data 3.2020.0512
Installing mime-types-data 3.2020.0512
Fetching net-http-digest_auth 1.4.1
Installing net-http-digest_auth 1.4.1
Fetching mini_portile2 2.4.0
Installing mini_portile2 2.4.0
Fetching ntlm-http 0.1.1
Installing ntlm-http 0.1.1
Fetching webrobots 0.1.2
Installing webrobots 0.1.2
Fetching method_source 1.0.0
Installing method_source 1.0.0
Fetching mustache 1.1.1
Installing mustache 1.1.1
Fetching parallel 1.19.2
Installing parallel 1.19.2
Fetching rainbow 3.0.0
Installing rainbow 3.0.0
Fetching sorbet-runtime 0.5.5866
Installing sorbet-runtime 0.5.5866
Fetching plist 3.5.0
Installing plist 3.5.0
Fetching rdiscount 2.2.0.2
Installing rdiscount 2.2.0.2 with native extensions
Fetching regexp_parser 1.7.1
Installing regexp_parser 1.7.1
Fetching rexml 3.2.4
Installing rexml 3.2.4
Fetching rspec-support 3.9.3
Installing rspec-support 3.9.3
Fetching ruby-progressbar 1.10.1
Installing ruby-progressbar 1.10.1
Fetching unicode-display_width 1.7.0
Installing unicode-display_width 1.7.0
Fetching ruby-macho 2.2.0
Installing ruby-macho 2.2.0
Fetching sorbet-static 0.5.5866 (x86_64-linux)
Installing sorbet-static 0.5.5866 (x86_64-linux)
Fetching thor 1.0.1
Installing thor 1.0.1
Fetching tzinfo 1.2.7
Installing tzinfo 1.2.7
Fetching parser 2.7.1.4
Installing parser 2.7.1.4
Fetching i18n 1.8.5
Installing i18n 1.8.5
Fetching elftools 1.1.2
Installing elftools 1.1.2
Fetching simplecov 0.18.5
Installing simplecov 0.18.5
Fetching commander 4.5.2
Installing commander 4.5.2
Fetching net-http-persistent 4.0.0
Installing net-http-persistent 4.0.0
Fetching unf 0.1.4
Installing unf 0.1.4
Fetching mime-types 3.3.1
Installing mime-types 3.3.1
Fetching nokogiri 1.10.10
Installing nokogiri 1.10.10 with native extensions
Fetching pry 0.13.1
Fetching parallel_tests 3.1.0
Installing pry 0.13.1
Installing parallel_tests 3.1.0
Fetching rspec-core 3.9.2
Fetching rspec-expectations 3.9.2
Installing rspec-core 3.9.2
Installing rspec-expectations 3.9.2
Fetching rspec-mocks 3.9.1
Fetching sorbet 0.5.5866
Installing sorbet 0.5.5866
Fetching rubocop-ast 0.3.0
Installing rubocop-ast 0.3.0
Fetching activesupport 6.0.3.2
Installing activesupport 6.0.3.2
Fetching patchelf 1.2.0
Installing patchelf 1.2.0
Fetching parlour 4.0.1
Installing parlour 4.0.1
Fetching domain_name 0.5.20190701
Installing domain_name 0.5.20190701
Fetching ronn 0.7.3
Installing ronn 0.7.3
Fetching codecov 0.2.5
Installing codecov 0.2.5
Fetching rspec-its 1.3.0
Installing rspec-its 1.3.0
Fetching rspec-retry 0.6.2
Installing rspec-retry 0.6.2
Fetching rubocop 0.88.0
Installing rubocop 0.88.0
Installing rspec-mocks 3.9.1
Fetching tapioca 0.4.1
Installing tapioca 0.4.1
Fetching http-cookie 1.0.3
Installing http-cookie 1.0.3
Fetching rspec 3.9.0
Fetching rubocop-performance 1.7.1
Installing rspec 3.9.0
Fetching rubocop-rspec 1.42.0
Installing rubocop-performance 1.7.1
Installing rubocop-rspec 1.42.0
Fetching rspec-wait 0.0.9
Installing rspec-wait 0.0.9
Fetching mechanize 2.7.6
Installing mechanize 2.7.6
Bundle complete! 21 Gemfile dependencies, 71 gems now installed.
Bundled gems are installed into `../../../../home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle`
Post-install message from i18n:

HEADS UP! i18n 1.1 changed fallbacks to exclude default locale.
But that may break your application.

If you are upgrading your Rails application from an older version of Rails:

Please check your Rails app for 'config.i18n.fallbacks = true'.
If you're using I18n (>= 1.1.0) and Rails (< 5.2.2), this should be
'config.i18n.fallbacks = [I18n.default_locale]'.
If not, fallbacks will be broken in your app by I18n 1.1.x.

If you are starting a NEW Rails application, you can ignore this notice.

For more info see:
https://github.com/svenfuchs/i18n/releases/tag/v1.1.0

Post-install message from sorbet:

  Thanks for installing Sorbet! To use it in your project, first run:

    bundle exec srb init

  which will get your project ready to use with Sorbet.
  After that whenever you want to typecheck your code, run:

    bundle exec srb tc

  For more docs see: https://sorbet.org/docs/adopting
🍺  /home/linuxbrew/.linuxbrew/Cellar/harfbuzz/2.7.0: 70 files, 7.4MB
Removing: /home/linuxbrew/.linuxbrew/Cellar/harfbuzz/2.6.8... (169 files, 13MB)
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  guile@2
```
